bundles/org.openhab.binding.dali/src/main/java/org/openhab/binding/dali/internal/handler/DaliRgbHandler.java

```diff
@@ -0,0 +1,130 @@
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DaliRgbHandler} handles commands for things of type RGB.
+ *
+ * @author Robert Schmid - Initial contribution
+ */
+@NonNullByDefault
+public class DaliRgbHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(DaliRgbHandler.class);
+
+    static final int Switch100Percent = 254;
+
+    public DaliRgbHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        initDeviceState();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE
+                && getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.BRIDGE_OFFLINE) {
+            initDeviceState();
+        } else if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    public void initDeviceState() {
+        Bridge bridge = getBridge();
+
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridge configured");
+        } else if (bridge.getStatus() == ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (CHANNEL_COLOR.equals(channelUID.getId())) {
+                List<Integer> outputs = List.of(
+                        ((BigDecimal) this.thing.getConfiguration().get(TARGET_ID_R)).intValueExact(),
+                        ((BigDecimal) this.thing.getConfiguration().get(TARGET_ID_G)).intValueExact(),
+                        ((BigDecimal) this.thing.getConfiguration().get(TARGET_ID_B)).intValueExact());
+
+                if (command instanceof HSBType) {
+                    PercentType[] rgb = ((HSBType) command).toRGB();
+
+                    for (int i = 0; i < 3; i++) {
+                        byte dimmValue = (byte) ((rgb[i].floatValue() * Switch100Percent) / 100);
+                        getBridgeHandler()
+                                .sendCommand(new DaliDAPCCommand(DaliAddress.Short(outputs.get(i)), dimmValue));
+                    }
+                } else if (command instanceof OnOffType) {
+                    if ((OnOffType) command == OnOffType.ON) {
+                        for (Integer output : outputs) {
+                            getBridgeHandler().sendCommand(
+                                    new DaliDAPCCommand(DaliAddress.Short(output), (byte) Switch100Percent));
+                        }
+                    } else {
```

Comment:
What happens if the command is REFRESH or INCREASE?

Reply:
I've now added handlers for refresh and increase/decrease commands.